bootloader/src/pyi_utils.c

```diff
@@ -740,6 +740,26 @@ pyi_utils_set_environment(const ARCHIVE_STATUS *status)
     return rc;
 }
 
+/*
+ * If the program is actived by systemd socket, LISTEN_PID is set to parent
+ * pid of the bootloader and should be changed to child pid.
+ */
+int
+set_systemd_env()
+{
+    const char * env_var = "LISTEN_PID";
+    if(pyi_getenv(env_var) != NULL) {
+        #define ULONG_STRING_SIZE (sizeof (unsigned long) * CHAR_BIT / 3 + 2)
```

Comment:
While still waiting for review by @codewarrior0: Where is this from? Please add a source. Thanks.